src/controllers/notificationsController.ts

```diff
@@ -0,0 +1,45 @@
+import { Request, Response } from 'express';
+import Notification from '../database/models/notification';
+import { sendInternalErrorResponse } from '../validations';
+import logger from '../logs/config';
+
+export const getNotifications = async (req: Request, res: Response): Promise<void> => {
+  try {
+    const { userId } = req.params;
+    const notifications = await Notification.findAll({ where: { userId } });
+    if (notifications.length === 0) {
+      logger.error('No notifications were found');
+      res.status(404).json({
+        ok: false,
+        message: 'No notifications that were found',
+      });
+      return;
+    }
+    res.status(200).json({ ok: true, data: notifications });
+  } catch (error) {
+    logger.error(error);
+    sendInternalErrorResponse(res, error);
+    return;
+  }
+};
+
+export const markNotificationAsRead = async (req: Request, res: Response): Promise<void> => {
+  const { id } = req.params;
+  const { isRead } = req.body;
```

Comment:
why are passing isRead in body? only notification id is enough to change isRead to true once user clicks on notification

Reply:
but I think this will help the user to track the current status of the notification!